driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/AggregatePublisherImpl.java

```diff
@@ -142,36 +142,33 @@
     private <E> Publisher<E> publishExplain(final Class<E> explainResultClass, @Nullable final ExplainVerbosity verbosity) {
         notNull("explainDocumentClass", explainResultClass);
         return getMongoOperationPublisher().createReadOperationMono(() ->
-                        asAggregateOperation().asAsyncExplainableOperation(verbosity,
+                        asAggregateOperation(1).asAsyncExplainableOperation(verbosity,
                                 getCodecRegistry().get(explainResultClass)),
                 getClientSession());
     }
 
     @Override
-    AsyncReadOperation<AsyncBatchCursor<T>> asAsyncReadOperation() {
+    AsyncReadOperation<AsyncBatchCursor<T>> asAsyncReadOperation(final int initialBatchSize) {
```

Comment:
Do we lose anything by not letting this be nullable, like it is in `getBatchSize()`?  Or does `0 == null`?

Reply:
A side but related question: is my understanding correct that we do not care about using boxed types and passing parameters by reference?

Reply:
We generally only box integral types when we need null to signal something, e.g. use the server's default value for this property.

Reply:
We will always set the batchSize - either the users configured batchSize or the initial demand

Reply:
Ah, I see.